Tests/YamsTests/TestHelper.swift

```diff
@@ -20,17 +19,13 @@ func timestamp(_ timeZoneHour: Int = 0,
                _ minute: Int? = nil,
                _ second: Int? = nil,
                _ fraction: Double? = nil ) -> Date {
-    let calendar = Calendar(identifier: .gregorian)
     let timeZone = TimeZone(secondsFromGMT: timeZoneHour * 60 * 60)
-    let nanosecond = fraction.map { Int($0 * Double(NSEC_PER_SEC)) }
-    let datecomponents = DateComponents(calendar: calendar, timeZone: timeZone,
-                          year: year, month: month, day: day,
-                          hour: hour, minute: minute, second: second, nanosecond: nanosecond)
-    // Using `DateComponents.date` causes crash on Linux
-    guard let date = NSCalendar(identifier: .gregorian)?.date(from: datecomponents) else {
-        fatalError("Never happen this")
-    }
-    return date
+    let dateComponents = DateComponents(calendar: gregorianCalendar, timeZone: timeZone,
+                                        year: year, month: month, day: day,
+                                        hour: hour, minute: minute, second: second)
+    guard let date = dateComponents.date else { fatalError("Never happen this") }
+    guard let fraction = fraction else { return date }
```

Comment:
Nit:

```swift
return date.addingTimeInterval(fraction ?? 0)
```

Reply:
Is the Swift optimizer smart enough to eliminate the code that adds `0` if `fraction` is `nil`?

Reply:
I changed to following:
```swift
    return fraction.map(date.addingTimeInterval) ?? date
```